sdk/main/src/AccountUpdateTransaction.cc

```diff
@@ -104,6 +104,14 @@ AccountUpdateTransaction& AccountUpdateTransaction::setAccountMemo(std::string_v
   return *this;
 }
 
+//-----
+AccountUpdateTransaction& AccountUpdateTransaction::clearAccountMemo()
+{
+  requireNotFrozen();
+  mAccountMemo = "";
```

Comment:
Since mAccountMemo is an optional, should the clear function not for example set it to std::nullopt. I think it is confusing having an optional string with "" as a value.

Reply:
I understand the confusion but setting `mAccountMemo` to `std::nullopt` wouldn't update the memo at all which is not desired behavior. Since `AccountUpdateTransaction` can be used to update as many or as little fields as we want, `std::optional` types are used to indicate if a value should be updated or not (i.e. if a `std::optional` field does not have a value, that field should not be updated) (this can be confirmed by looking at the `build()` function). So we want to make sure `mAccountMemo` is initialized with a value so that it gets built into the `AccountUpdateTransaction`'s corresponding protobuf when it gets executed, but we want it to be empty i.e. cleared. If `mAccountMemo` was set to `std::nullopt` here, then no change would be built into the protobuf and the account memo would not actually be cleared. Does that make sense?

Reply:
I understand now. Thanks!